packages/create-sitecore-jss/src/templates/nextjs-personalize/src/components/CdpIntegrationScript.tsx

```diff
@@ -1,89 +1,66 @@
-import { useSitecoreContext } from '@sitecore-jss/sitecore-jss-nextjs';
-import Script from 'next/script';
+import { useSitecoreContext, CdpHelper } from '@sitecore-jss/sitecore-jss-nextjs';
 import { useEffect } from 'react';
 import config from 'temp/config';
+import { init } from '@sitecore/engage';
 
-declare const _boxeverq: { (): void }[];
-declare const Boxever: Boxever;
-
-interface Boxever {
-  getID(): string;
-  eventCreate(data: BoxeverViewEventArgs, callback: () => void, format: string): void;
-}
-
-interface BoxeverViewEventArgs {
-  browser_id: string;
-  channel: string;
-  type: string;
-  language: string;
-  page: string;
-  pos: string;
-}
-
-function createPageView(locale: string, routeName: string) {
-  // POS must be valid in order to save events (domain name might be taken but it must be defined in CDP settings)
-  const pointOfSale =
-    process.env.NEXT_PUBLIC_CDP_POINTOFSALE || window.location.host.replace(/^www\./, '');
-
-  _boxeverq.push(function () {
-    const pageViewEvent: BoxeverViewEventArgs = {
-      browser_id: Boxever.getID(),
-      channel: 'WEB',
-      type: 'VIEW',
-      language: locale,
-      page: routeName,
-      pos: pointOfSale,
-    };
-
-    Boxever.eventCreate(
-      pageViewEvent,
-      function () {
-        /*empty callback*/
-      },
-      'json'
-    );
-  });
-}
-
+/**
+ * This is the CDP integration script which uses the Sitecore Engage SDK.
+ * It is used to enable page view events.
+ * See Sitecore Engage SDK documentation for details.
+ * https://www.npmjs.com/package/@sitecore/engage
+ */
 const CdpIntegrationScript = (): JSX.Element => {
```

Comment:
You should probably rename this, right?

Reply:
I assume you mean switching to something like `EngageIntegrationScript`? We discussed this a bit but that seems more of an implementation detail. Whereas this is meant to indicate integration with Sitecore _CDP_ product.

Reply:
Sorry, my comment was a bit cryptic. No, I meant that this is no longer loading a script, so that last part of the name could be misleading.

In terms of mentioning "engage" or "CDP", I'd probably avoid mentioning either and go for something more descriptive. I know that tracking is a "forbidden" term, otherwise I'd suggest something that matches the comments like "PageViewEventTracker" or similar.

Reply:
Ah, this is a good point. Even though this is included through our [Scripts](https://github.com/Sitecore/jss/pull/1166/files#diff-0dae17efb21ff6cfc0b89abf0a6bf169fecc27e153cd7318d2de05d1dd2ad27c) abstraction, it doesn't need to use this term itself. I went with "CdpPageView" (to avoid "tracking" as you say) :)